amethyst_network/src/server/server_socket_event.rs

```diff
@@ -0,0 +1,44 @@
+use laminar::error::NetworkError;
+use laminar::{Event, Packet};
+
+/// Net event which occurred on the network.
+pub enum ServerSocketEvent {
+    /// event containing a packet with received data
+    Packet(Packet),
+    /// event containing an error that has occurred in the network
+    Error(NetworkError),
+    /// events that can happen with a client
+    ClientEvent(ClientEvent),
+    /// event used for defaults
+    Empty,
+}
+
+impl Clone for ServerSocketEvent {
+    fn clone(&self) -> Self {
+        ServerSocketEvent::Empty
```

Comment:
This doesn't seem right, seems like a default impl but this is for clone?

Reply:
Clone is used in a vector allocation: `

 let mut buffer = vec![ServerSocketEvent::Empty; receiver.max_throughput];`

to this buffer, incoming events will be written.

Reply:
I don't understand what you mean here? Clone should be to kinda copy the item not construct a new one?

Reply:
Clone should indeed copy the internal state of whatever is being cloned. I think this would be default, not clone.